tests/unit/test_loader.py

```diff
@@ -310,40 +317,47 @@ def test_context(self):
         self.assertEqual(self.loader._dict, {})
         # get something, and make sure its a func
         func = self.loader["test.ping"]
-        with patch.dict(func.__globals__["__context__"], {"foo": "bar"}):
+        with salt.loader_context.loader_context(self.loader):
+            with patch.dict(func.__globals__["__context__"], {"foo": "bar"}):
+                self.assertEqual(
+                    self.loader["test.echo"].__globals__["__context__"]["foo"], "bar"
+                )
+                self.assertEqual(
+                    self.loader["grains.get"].__globals__["__context__"]["foo"], "bar"
+                )
```

Comment:
**question**: does it make a difference if this passes by adding `loader_context = MagicMock()`? I'm pretty sure that the answer is "no" IIUC.

Reply:
That is to be expected. The difference is that when we include these loader changes the `loader_context` call needs to be included for the test to pass. On master the test would pass by just removing line 320 or mocking the loader_context call.